distribution/kpkginstall/runtest.sh

```diff
@@ -38,14 +44,29 @@ function set_package_name()
       PACKAGE_NAME=kernel
   fi
 
+  # Append "-debug" if we were asked to install the debug kernel.
+  if [[ "${KPKG_INSTALL_DEBUG:-}" == "yes" ]]; then
+    echo "Debug kernel was requested -- appending -debug to package name"
+    PACKAGE_NAME=${PACKAGE_NAME}-debug
+  fi
+
+  # Write the PACKAGE_NAME to a file in /tmp so we have it after reboot.
+  echo -n "${PACKAGE_NAME}" | tee -a /tmp/kpkginstall/KPKG_PACKAGE_NAME
+
   echo "Package name is ${PACKAGE_NAME}" | tee -a ${OUTPUTFILE}
 }
 
 function get_kpkg_ver()
 {
+  # Recover the saved package name from /tmp/KPKG_KVER if it exists.
+  if [ -f "/tmp/kpkginstall/KPKG_KVER" ]; then
+    KVER=$(cat /tmp/kpkginstall/KPKG_KVER)
+    return
+  fi
+
   if [[ "${KPKG_URL}" =~ .*\.tar\.gz ]] ; then
     declare -r kpkg=${KPKG_URL##*/}
-    tar tf "$kpkg" | sed -ne '/^boot\/vmlinu[xz]-[1-9]/ {s/^[^-]*-//p;q}; $Q1'
+    KVER=$(tar tf "$kpkg" | sed -ne '/^boot\/vmlinu[xz]-[1-9]/ {s/^[^-]*-//p;q}; $Q1')
```

Comment:
What is `$Q1`? Is that a sed thing? I don't remember having seen it before.

Reply:
Not me -- Artem (the original author) used it and I kept it there with a "it works, don't touch it" logic. However, doing some research it looks like it means "if we got this far quit with exit code 1". So the code tries a few matches to get the kernel version and if none of them match, returns an unsuccessful return code.

The broken down logic looks like `$` -- got to last line; `Q` -- command to quit; `1` -- parameter to quit command